### PR TITLE
Get legend request

### LINKED ISSUE #9163: GetLegend requests
SLD enabled WMS servers can support GetLegendGraphic requests. I will need this for a project. If desired I can implement a function similar to the getGetFeatureInfoUrl of the ImageWMS class for openlayers. That would be a getGetLegendGraphicUrl

Some more information on the topic can be found here: http://suite.opengeo.org/ee/docs/4.5/geoserver/services/wms/get_legend_graphic/legendgraphic.html and in the sld specification chapter 10: http://portal.opengeospatial.org/files/?artifact_id=22364

### PR DESCRIPTION
**Pull request by simonseyock**
Fixes #9163 

It is based on @KaiVolland s work #9398

### CODE REVIEW

**Comment by simonseyock** [comment]
@ahocevar what do you think of this pull request?

File: src/ol/source/ImageWMS.js [hunk 1]
```diff
@@ -191,6 +191,43 @@ class ImageWMS extends ImageSource {
       1, sourceProjectionObj || projectionObj, baseParams);
   }
 
+  /**
+   * Return the GetLegendGraphic URL for the passed resolution.
+   * Return `undefined` if the GetLegendGraphic URL cannot be constructed.
+   * @param {!number} resolution Resolution. If set to undefined `SCALE`
```
Comment (jahow) [inline src/ol/source/ImageWMS.js]: Optional parameters should be using the `{Type} [name]` syntax now, I know this is not consistent with the rest of the file but this is the correct JSDoc syntax nowadays.

See for example this recently merged PR: https://github.com/openlayers/openlayers/pull/9751

Comment (marcjansen) [reply src/ol/source/ImageWMS.js]: I'll do this in a follow-up. Thanks for you review.

**Comment by marcjansen** [comment]
Thanks for the work on this @KaiVolland , @simonseyock and @jahow.

File: src/ol/source/ImageWMS.js (hunk 1 above)
Comment (marcjansen) [reply src/ol/source/ImageWMS.js]: See https://github.com/openlayers/openlayers/pull/9792

**Pull request merged**